src/UsgsAstroProjectedLsSensorModel.cpp

```diff
@@ -0,0 +1,799 @@
+std::string UsgsAstroProjectedLsSensorModel::getParameterName(int index) const {
+  return PARAMETER_NAME[index];
+}
+
+std::string UsgsAstroProjectedLsSensorModel::getParameterUnits(int index) const {
+  // All parameters are meters or scaled to meters
+  return "m";
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::getNumParameters
+//***************************************************************************
+int UsgsAstroProjectedLsSensorModel::getNumParameters() const {
+  return UsgsAstroProjectedLsSensorModel::NUM_PARAMETERS;
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::getParameterType
+//***************************************************************************
+csm::param::Type UsgsAstroProjectedLsSensorModel::getParameterType(int index) const {
+  return m_parameterType[index];
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::setParameterType
+//***************************************************************************
+void UsgsAstroProjectedLsSensorModel::setParameterType(int index,
+                                              csm::param::Type pType) {
+  m_parameterType[index] = pType;
+}
+
+//---------------------------------------------------------------------------
+// Sensor Model Information
+//---------------------------------------------------------------------------
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::getPedigree
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getPedigree() const {
+  return "USGS_LINE_SCANNER";
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::getImageIdentifier
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getImageIdentifier() const {
+  return m_imageIdentifier;
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::setImageIdentifier
+//***************************************************************************
+void UsgsAstroProjectedLsSensorModel::setImageIdentifier(const std::string& imageId,
+                                                csm::WarningList* warnings) {
+  // Image id should include the suffix without the path name
+  m_imageIdentifier = imageId;
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::getSensorIdentifier
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getSensorIdentifier() const {
+  return m_sensorIdentifier;
+}
```

Comment:
Im guessing a lot of the getters are not inherited?

Reply:
No they are, I just didn't remove them, they are gone now